diff --git a/drivers/gpu/drm/radeon/si_dpm.c b/drivers/gpu/drm/radeon/si_dpm.c
index 10191b9..8409dea 100644
--- a/drivers/gpu/drm/radeon/si_dpm.c
+++ b/drivers/gpu/drm/radeon/si_dpm.c
@@ -3010,16 +3010,18 @@ static void si_apply_state_adjust_rules(struct radeon_device *rdev,
 			max_mclk = 120000;
 	} else if (rdev->family == CHIP_VERDE) {
 		if ((rdev->pdev->revision == 0x81) ||
-		    (rdev->pdev->revision == 0x83) ||
 		    (rdev->pdev->revision == 0x87) ||
 		    (rdev->pdev->device == 0x6820) ||
-		    (rdev->pdev->device == 0x6821) ||
 		    (rdev->pdev->device == 0x6822) ||
 		    (rdev->pdev->device == 0x6823) ||
 		    (rdev->pdev->device == 0x682A) ||
 		    (rdev->pdev->device == 0x682B)) {
 			max_sclk = 75000;
 			max_mclk = 80000;
+		} else if (rdev->pdev->device == 0x6821 &&
+				rdev->pdev->revision == 0x83) {
+			max_sclk = 80000;
+			max_mclk = 112500;
 		}
 	} else if (rdev->family == CHIP_OLAND) {
 		if ((rdev->pdev->revision == 0xC7) ||
